### ISSUE TITLE
please update src/stage0 for beta branch

### ISSUE THREAD

**Issue by stefson** [rust-lang/rust#126474]
hello everyone, 

I downloaded the rustc-beta src tarball and tried to compile a rustc-1.80.0_beta version from it, but it failed as the build process still tries to use a nightly compiler for that: 

```
running sanity check
learning about cargo
thread 'main' panicked at src/core/download.rs:635:74:
src/stage0 doesn't contain a checksum for dist/2024-05-29/rustfmt-nightly-x86_64-gentoo-linux-musl.tar.xz. Pre-built artifacts might not be available for this target at this time, see https://doc.rust-lang.org/nightly/rustc/platform-support.html for more information.
stack backtrace:
   0:     0x55cad4147de5 - <std::sys_common::backtrace::_print::DisplayBacktrace as core::fmt::Display>::fmt::h11b410cd49fe3fb3
   1:     0x55cad417422b - core::fmt::write::h4648a7e0bcc98ee6
   2:     0x55cad414369f - std::io::Write::write_fmt::h57cbde3af46ae559
   3:     0x55cad4147bbe - std::sys_common::backtrace::print::hfbdbcb2103cd101a
   4:     0x55cad4149899 - std::panicking::default_hook::{{closure}}::h1ac7fb5046e230ee
   5:     0x55cad41495dd - std::panicking::default_hook::hcec563d0a338315d
   6:     0x55cad4149d33 - std::panicking::rust_panic_with_hook::hcfd0a8a68ea11baa
   7:     0x55cad4149c14 - std::panicking::begin_panic_handler::{{closure}}::h9597682dd2115ee8
   8:     0x55cad41482a9 - std::sys_common::backtrace::__rust_end_short_backtrace::h39df4c4f1448b7c2
   9:     0x55cad4149947 - rust_begin_unwind
  10:     0x55cad3438f33 - core::panicking::panic_fmt::ha8ee10af32fb35a3
  11:     0x55cad4172a0c - core::panicking::panic_display::h7960d3e409d71a90
  12:     0x55cad3438efc - core::option::expect_failed::ha0f5a409c07f0639
  13:     0x55cad36cd9ca - core::option::Option<T>::expect::h7eb261c34cee2e11
                               at /rustc/129f3b9964af4d4a709d1383930ade12dfe7c081/library/core/src/option.rs:895:21
  14:     0x55cad353acce - bootstrap::core::download::<impl bootstrap::core::config::config::Config>::download_component::h3f8713af32d8c2ef
                               at /var/tmp/portage/dev-lang/rust-1.80.0_beta/work/rustc-beta-src/src/bootstrap/src/core/download.rs:635:26
  15:     0x55cad353709d - bootstrap::core::download::<impl bootstrap::core::config::config::Config>::maybe_download_rustfmt::h720da2c594f7285f
                               at /var/tmp/portage/dev-lang/rust-1.80.0_beta/work/rustc-beta-src/src/bootstrap/src/core/download.rs:429:9
  16:     0x55cad352e852 - bootstrap::core::config::config::Config::initial_rustfmt::h94c9c402330895d3
                               at /var/tmp/portage/dev-lang/rust-1.80.0_beta/work/rustc-beta-src/src/bootstrap/src/core/config/config.rs:2267:28
  17:     0x55cad3453a5d - bootstrap::Build::initial_rustfmt::hdd64aad5de38d0bc
                               at /var/tmp/portage/dev-lang/rust-1.80.0_beta/work/rustc-beta-src/src/bootstrap/src/lib.rs:291:17
  18:     0x55cad34480b1 - bootstrap::Build::build::he4b80b01cfb6391e
                               at /var/tmp/portage/dev-lang/rust-1.80.0_beta/work/rustc-beta-src/src/bootstrap/src/lib.rs:660:18
  19:     0x55cad3440003 - bootstrap::main::h85310da40514247b
                               at /var/tmp/portage/dev-lang/rust-1.80.0_beta/work/rustc-beta-src/src/bootstrap/src/bin/main.rs:79:5
  20:     0x55cad3442733 - core::ops::function::FnOnce::call_once::hcd07113498239920
                               at /rustc/129f3b9964af4d4a709d1383930ade12dfe7c081/library/core/src/ops/function.rs:250:5
  21:     0x55cad3441c16 - std::sys_common::backtrace::__rust_begin_short_backtrace::h1ec5de0a56f1ba90
                               at /rustc/129f3b9964af4d4a709d1383930ade12dfe7c081/library/std/src/sys_common/backtrace.rs:155:18
  22:     0x55cad343a3d9 - std::rt::lang_start::{{closure}}::ha1b720be0eb0ec7c
                               at /rustc/129f3b9964af4d4a709d1383930ade12dfe7c081/library/std/src/rt.rs:159:18
  23:     0x55cad413a16d - std::rt::lang_start_internal::h539634803847187d
  24:     0x55cad343a3b7 - std::rt::lang_start::h480a750233594790
                               at /rustc/129f3b9964af4d4a709d1383930ade12dfe7c081/library/std/src/rt.rs:158:17
  25:     0x55cad3441a6e - main
  26:     0x7f4d6fd29f8b - <unknown>
Traceback (most recent call last):
  File "/var/tmp/portage/dev-lang/rust-1.80.0_beta/work/rustc-beta-src/./x.py", line 50, in <module>
    bootstrap.main()
  File "/var/tmp/portage/dev-lang/rust-1.80.0_beta/work/rustc-beta-src/src/bootstrap/bootstrap.py", line 1191, in main
    bootstrap(args)
  File "/var/tmp/portage/dev-lang/rust-1.80.0_beta/work/rustc-beta-src/src/bootstrap/bootstrap.py", line 1167, in bootstrap
    run(args, env=env, verbose=build.verbose, is_bootstrap=True)
  File "/var/tmp/portage/dev-lang/rust-1.80.0_beta/work/rustc-beta-src/src/bootstrap/bootstrap.py", line 186, in run
    raise RuntimeError(err)
RuntimeError: failed to run: /var/tmp/portage/dev-lang/rust-1.80.0_beta/work/rustc-beta-src/build/bootstrap/debug/bootstrap build -vv --config=/var/tmp/portage/dev-lang/rust-1.80.0_beta/work/rustc-beta-src/config.toml -j12
```

the stage0 process used to be a json file, this has propably changed with rust-1.80.0-nightly - will there be an update to that stage0 file in the near future, or are there some deeper changes in the process of fabricating the src tarballs?

**Comment by stefson** [comment]
forgot to add my config.toml: 

```
[llvm]
download-ci-llvm = false
optimize = true
release-debuginfo = false
assertions = false
ninja = true
targets = "X86"
experimental-targets = ""
link-shared = false
[build]
build = "x86_64-gentoo-linux-musl"
host = ["x86_64-gentoo-linux-musl"]
target = ["x86_64-gentoo-linux-musl"]
cargo = "/usr/lib/rust/1.79.0/bin/cargo"
rustc = "/usr/lib/rust/1.79.0/bin/rustc"
docs = false
compiler-docs = false
submodules = false
python = "python3.12"
locked-deps = true
vendor = true
extended = true
tools = ["rustfmt","rust-demangler","cargo","rustdoc"]
verbose = 2
sanitizers = false
profiler = true
cargo-native-static = false
[install]
prefix = "/usr/lib/rust/1.80.0_beta"
sysconfdir = "etc"
docdir = "share/doc/rust"
bindir = "bin"
libdir = "lib"
mandir = "share/man"
[rust]
# https://github.com/rust-lang/rust/issues/54872
codegen-units-std = 1
optimize = true
debug = false
debug-assertions = false
debuginfo-level-rustc = 0
backtrace = true
incremental = false
default-linker = "x86_64-gentoo-linux-musl-gcc"
parallel-compiler = false
channel = "stable"
rpath = false
verbose-tests = true
optimize-tests = true
codegen-tests = true
dist-src = false
# https://github.com/rust-lang/rust/issues/117885
remap-debuginfo = true
lld = false
backtrace-on-ice = true
jemalloc = false
# See https://github.com/rust-lang/rust/issues/121124
lto = "off"
[dist]
src-tarball = false
compression-formats = ["xz"]
[target.x86_64-gentoo-linux-musl]
cc = "x86_64-gentoo-linux-musl-gcc"
cxx = "x86_64-gentoo-linux-musl-g++"
linker = "x86_64-gentoo-linux-musl-gcc"
ar = "x86_64-gentoo-linux-musl-ar"
crt-static = false
```

I'm building offline

**Comment by cuviper** [comment]
This should be addressed by #126453, as we drop rustfmt-nightly when we bump stage0 on the beta branch.

**Comment by workingjubilee** [comment]
Landed. Should be fixed.
